### ISSUE TITLE
Request: Add option to indent 'break' same level as 'case'

### ISSUE THREAD

**Issue by chrisemb** [uncrustify/uncrustify#2281]
It would be great if uncrustify had an option to have `break` in a `switch` body indented to the same level as `case`:
```c
    int foo(int op)
    {
        switch (op)
        {
            case 1:
                do_something();
            break;
            case 2:
                do_something_else();
            case 3:
                do_something_different();
                do_something_more();
            break;
        }
        return -1;
    }
```
This helps to find a missing `break` and to see if a `case` falls through to the next one.

This is required, for example, in the safety-oriented coding standard of the Barr Group ([rule 8.3.a](https://barrgroup.com/Embedded-Systems/Books/Embedded-C-Coding-Standard/Statement-Rules/Switch-Statements)).

This option can also be found in the CDT formatter of Eclipse (indent_breaks_compare_to_cases=false).

**Comment by chrisemb** [comment]
An apparently similar request was made earlier: #704

That request contained the statements `return`, `goto` and `exit` in addition to the `break`. It also had an example C source, including cases where those statements would keep their deeper indentation level (within `if` statements).

While a comment indicates that the request had been implemented, I cannot find any option to actually enable that behaviour.

Either the `break` only request #2281 or the more general #704 wold be fine for me.

The indentation of a `break` inside an `if` was not specified in #2281 but should be handled as suggested in #704:

```c
    int foo(int op)
    {
        switch (op)
        {
            case 1:
                do_something();
            break;
            case 2:
                do_something_else();
            case 3:
                if (do_something_different())
                {
                    break;
                }
                do_something_more();
            break;
        }
        return -1;
    }
```

**Comment by guy-maurel** [comment]
The new option _indent_switch_break_with_case = true_
does what you are looking for.
The current git version works pretty fine.
  https://github.com/uncrustify/uncrustify
Please test it and report!

**Comment by chrisemb** [comment]
Thank you.
I checked the new option with uncrustify 0.70.1.
It is basically fine but inside an 'if' a 'break' should be indented like surrounding code.
Current result:
```c
    int foo (int op)
    {
        switch (op)
        {
            case 1:
                do_something();
            break;
            case 2:
                do_something_else();
            case 3:
                if (do_something_different())
                {
                    do_this();
                break;  // this should be indented like the surrounding code
                }
                do_something_more();
            break;
        }
        return -1;
    }
```

For the intended indention please also see my post above.
The same wrong indentation happens to the example of @AmokHuginnsson in #1692.

**Comment by guy-maurel** [comment]
uncrustify cannot  today distinguish which block contains the break.
A new structur is necessary. Please wait a little.

**Comment by guy-maurel** [comment]
Ref. #2567 
Some prepares are necessary.

**Comment by chrisemb** [comment]
I follow the process occasionally and appreciate it very much to see the progress, thanks.

**Comment by gmaurel** [comment]
Please, test at new.

**Comment by chrisemb** [comment]
Thanks, I first tested with

two input files
-	Issue_1692.cpp (from the project) and
-	my_infile.c (which is a heavily misaligned sample file, mainly made up from code mentioned in earlier feature requests for indentation of `break`.)

and two configuration files
-	Issue_1692.cfg (from the project) and
-	my_conf.cfg (a preliminary version of the file I want to use in the future).

This allows for four combinations to generate output files:
- out1.cpp (Issue_1692.cpp with Issue_1692.cfg); fine, not attached
- out2.cpp (Issue_1692.cpp with my_conf.cfg); fine, not attached
- out3.c (my_infile.c with Issue_1692.cfg); all `break`s are fine except for the `break` in line 140! But the equivalent `break` in line 178 of out4.c, below, is fine
- out4.c (my_infile.c with my_conf.cfg); all `break`s are fine, except for the `break` in line 16 and the `break` in line 120, which is misaligned even though the construct is exactly the one of Issue_1692.cpp; equivalent `break`s in out3.c are fine; Changing my_infile.c just by removing spaces in line 15 (as in my_infile2.c) results in line 16 of out42.c being fine.

Could you please check for the reasons?
[files.zip](https://github.com/uncrustify/uncrustify/files/4054008/files.zip)

**Comment by chrisemb** [comment]
It seems I accidentally closed this issue

**Comment by guy-maurel** [comment]
Thank for the big test data.
Please, test again and report.

**Comment by chrisemb** [comment]
Thanks, great. I also tested with some real world source files and it works like a charm.

**Comment by chrisemb** [comment]
Vielen Dank für Ihre E-Mail. 

Leider bin ich zur Zeit nicht im Büro. Ich werde am Montag, 2. März zurück sein. In dringenden Fällen wenden Sie sich bitte telefonisch an die zentrale Rufnummer. Bitte beachten Sie, dass Ihre E-Mail nicht automatisch weitergeleitet wird.


Mit freundlichen Grüßen
Christian Ening
Dipl.-Ing.




Thank you for your e-mail. 


Unfortunately, I'm currently out of office. I'll be back on Monday, March 2nd. In urgent cases, please call the central telephone number. Please note that your e-mail will not be forwarded automatically.

With kind regards,
Christian Ening
_________________________________ 

ser elektronik GmbH
Zeissweg 6
59519 Möhnesee

Telefon: +49 2924 9717 0
Telefax: +49 2924 9717 45
http://www.ser-elektronik.de

Geschäftsführer:
Ferdinand Eickhoff

Handelsgericht Arnsberg HRB 5458
Ust.-IdNr.: DE126637722
St.-Nr.: 343/5782/2052
Closed #2281.

-- 
You are receiving this because you modified the open/close state.
Reply to this email directly or view it on GitHub:
https://github.com/uncrustify/uncrustify/issues/2281#event-3073309474
